nova/quota.py

```diff
@@ -184,18 +184,19 @@ def _get_usages(self, context, resources, project_id, user_id=None):
             # such as AbsoluteResources.
             if not isinstance(resource, CountableResource):
                 continue
-            if resource.name in usages or 'instances' in usages and \
-                    resource.name.startswith('instances_'):
+            if resource.name in usages or \
```

Comment:
For the commit message. (I'm just putting this here, so we can have a thread about it.)

> quota: Introduce SAPCustomResource
>
> When we add our own resources for quota, e.g. the
> "instances_$FLAVOR_NAME" resources for counting instances of a specific
> flavor, these resources usually don't come up in the usage-count unless
> a project uses them. Since we use the same function for counting the
> instances_* resources and the cores/ram, and these always get set, we

"these" -- I'm fairly certain you mean "the latter", i.e. cores/ram, but it's not clear.

> can check for them to see if we have to run our resource-counting
> function (again).

Reply:
fixed